storage-node/src/commands/operator/set-metadata.ts

```diff
@@ -45,16 +57,55 @@ export default class OperatorSetMetadata extends ApiCommandBase {
 
   async run(): Promise<void> {
     const { flags } = this.parse(OperatorSetMetadata)
-    const { workerId, bucketId, jsonFile, endpoint } = flags
+    const { workerId, bucketId, jsonFile, endpoint, operationalStatus: statusType } = flags
+
+    let operationalStatus: INodeOperationalStatusMetadata = {}
+    switch (statusType) {
+      case 'Normal': {
+        operationalStatus = { status: NodeOperationalStatusMetadata.OperationalStatus.NORMAL }
+        break
+      }
+      case 'NoService': {
+        operationalStatus = { status: NodeOperationalStatusMetadata.OperationalStatus.NO_SERVICE }
+        break
+      }
+      case 'NoServiceFrom': {
+        operationalStatus = {
+          status: NodeOperationalStatusMetadata.OperationalStatus.NO_SERVICE,
+          noServiceFrom: (await this.datePrompt({ message: 'Enter No Service period start date' })).toISOString(),
+        }
+        break
+      }
+      case 'NoServiceDuring': {
+        operationalStatus = {
+          status: NodeOperationalStatusMetadata.OperationalStatus.NO_SERVICE,
+          noServiceFrom: (await this.datePrompt({ message: 'Enter No Service period start date' })).toISOString(),
+          noServiceTo: (await this.datePrompt({ message: 'Enter No Service period end date' })).toISOString(),
+        }
+      }
+    }
 
     const validation = new ValidationService()
-    const metadata: IStorageBucketOperatorMetadata = jsonFile
-      ? validation.validate('OperatorMetadata', JSON.parse(fs.readFileSync(jsonFile).toString()))
-      : { endpoint }
+    let metadata: IStorageBucketOperatorMetadata
+    if (jsonFile) {
+      const input = validation.validate('OperatorMetadata', JSON.parse(fs.readFileSync(jsonFile).toString()))
+      metadata = {
+        ...input,
+        operationalStatus: {
+          ...input.operationalStatus,
+          status:
+            input.operationalStatus?.status === 'Normal'
+              ? NodeOperationalStatusMetadata.OperationalStatus.NORMAL
+              : NodeOperationalStatusMetadata.OperationalStatus.NO_SERVICE,
+        },
+      }
```

Comment:
I think if somebody will try to set metadata without operational status then this will set it to `NO_SERVICE`?

Reply:
Addressed in [6ff4531](https://github.com/Joystream/joystream/pull/4793/commits/6ff45319432ff711ce35e281f01b0fba889089a3)